src/OrchardCore/OrchardCore.DisplayManagement/Zones/ZoneShapes.cs

```diff
@@ -105,15 +105,18 @@ public async Task<IHtmlContent> ContentZone(dynamic DisplayAsync, dynamic Shape,
             }
             else
             {
-                // Evaluate for cards.
-                var cardGrouping = await ShapeFactory.CreateAsync("CardGrouping", Arguments.From(
-                    new
-                    {
-                        Grouping = groupings[0],
-                        ContentItem = Shape.ContentItem
-                    }));
+               if ( groupings.FirstOrDefault() != null){
```

Comment:
Yeah I can see how it's possible to still end up with an empty zone, if you've removed the `TitlePart` manually for example.

The better change here (I think I originally had it in, and then removed it, would be

```
else if (groupings.Count == 1)
{
  var cardGrouping ... etc
}
```